nuxeo-services/nuxeo-platform-web-common/src/main/java/org/nuxeo/ecm/platform/web/common/requestcontroller/service/NuxeoCorsFilterDescriptor.java

```diff
@@ -74,15 +79,37 @@
     @XNode("@maxAge")
     protected int maxAge = -1;
 
-    protected String pattern = "";
+    protected Pattern pattern;
 
     @XNode("pattern")
-    public void setPattern(String pattern) {
-        this.pattern = Framework.expandVars(pattern);
+    public void setPattern(String patternString) {
+        patternString = Framework.expandVars(patternString);
+        if (!StringUtils.isBlank(patternString)) {
+            pattern = Pattern.compile(patternString);
+        }
+    }
+
+    protected volatile CORSFilter filter;
+
+    public CORSFilter getFilter() {
+        if (filter == null) {
+            synchronized(this) {
```

Comment:
I wouldn't use `synchronized(this)`. I prefer to use an internal `new Object()` as lock

Reply:
This is the standard double-checked locking idiom and using `this` is fine.

https://wiki.sei.cmu.edu/confluence/display/java/LCK10-J.+Use+a+correct+form+of+the+double-checked+locking+idiom#LCK10-J.Useacorrectformofthedouble-checkedlockingidiom-CompliantSolution(Volatile)